src/modules/directory.rs

```diff
@@ -223,6 +246,15 @@ mod tests {
         assert_eq!(output, "C:/");
     }
 
+    #[test]
+    fn substitute_prefix_and_middle() {
+        let full_path = "/absolute/path/foo/bar/baz";
+        let substitutions = vec![["/absolute/path", ""].to_vec(), ["/bar/", "/"].to_vec()];
```

Comment:
Is `vec![[].to_vec(), ...]` the best way to write this vector-of-vectors in code? Related, does taking a `&[Vec<&str>]` parameter to pass a `&Vec<Vec<&str>>` make sense?  I was mostly following compiler suggestions here.